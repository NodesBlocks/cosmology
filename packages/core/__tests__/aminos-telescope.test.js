// @ts-nocheck
import { osmosis } from 'osmojs';

it('aminos', async () => {
  const additions = {
    ...osmosis.gamm.v1beta1.AminoConverter,
    ...osmosis.lockup.AminoConverter,
    ...osmosis.superfluid.AminoConverter
  };
  expect(additions).toMatchSnapshot();
});

it('lockTokens.fromAmino', async () => {
  const fromAmino = osmosis.lockup.AminoConverter[
    '/osmosis.lockup.MsgLockTokens'
  ].fromAmino({
    owner: 'osmo1addresshere',
    coins: [
      {
        denom: 'mydenom',
        amount: '1000'
      }
    ],
    duration: {
      seconds: '100',
      nanos: 0
    }
  });
  expect(fromAmino).toMatchSnapshot();
});
it('lockTokens.toAmino', async () => {
  const toAmino = osmosis.lockup.AminoConverter[
    '/osmosis.lockup.MsgLockTokens'
  ].toAmino({
    owner: 'osmo1addresshere',
    coins: [
      {
        denom: 'mydenom',
        amount: '1000'
      }
    ],
    duration: {
      seconds: '100',
      nanos: 0
    }
  });
  expect(toAmino).toMatchSnapshot();
});
