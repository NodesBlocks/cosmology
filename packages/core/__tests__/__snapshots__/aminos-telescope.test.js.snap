// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`aminos 1`] = `
Object {
  "/osmosis.gamm.v1beta1.MsgExitPool": Object {
    "aminoType": "osmosis/gamm/exit-pool",
    "fromAmino": [Function],
    "toAmino": [Function],
  },
  "/osmosis.gamm.v1beta1.MsgExitSwapExternAmountOut": Object {
    "aminoType": "osmosis/gamm/exit-swap-extern-amount-out",
    "fromAmino": [Function],
    "toAmino": [Function],
  },
  "/osmosis.gamm.v1beta1.MsgExitSwapShareAmountIn": Object {
    "aminoType": "osmosis/gamm/exit-swap-share-amount-in",
    "fromAmino": [Function],
    "toAmino": [Function],
  },
  "/osmosis.gamm.v1beta1.MsgJoinPool": Object {
    "aminoType": "osmosis/gamm/join-pool",
    "fromAmino": [Function],
    "toAmino": [Function],
  },
  "/osmosis.gamm.v1beta1.MsgJoinSwapExternAmountIn": Object {
    "aminoType": "osmosis/gamm/join-swap-extern-amount-in",
    "fromAmino": [Function],
    "toAmino": [Function],
  },
  "/osmosis.gamm.v1beta1.MsgJoinSwapShareAmountOut": Object {
    "aminoType": "osmosis/gamm/join-swap-share-amount-out",
    "fromAmino": [Function],
    "toAmino": [Function],
  },
  "/osmosis.gamm.v1beta1.MsgSwapExactAmountIn": Object {
    "aminoType": "osmosis/gamm/swap-exact-amount-in",
    "fromAmino": [Function],
    "toAmino": [Function],
  },
  "/osmosis.gamm.v1beta1.MsgSwapExactAmountOut": Object {
    "aminoType": "osmosis/gamm/swap-exact-amount-out",
    "fromAmino": [Function],
    "toAmino": [Function],
  },
  "/osmosis.lockup.MsgBeginUnlocking": Object {
    "aminoType": "osmosis/lockup/begin-unlocking",
    "fromAmino": [Function],
    "toAmino": [Function],
  },
  "/osmosis.lockup.MsgBeginUnlockingAll": Object {
    "aminoType": "osmosis/lockup/begin-unlocking-all",
    "fromAmino": [Function],
    "toAmino": [Function],
  },
  "/osmosis.lockup.MsgExtendLockup": Object {
    "aminoType": "osmosis/lockup/extend-lockup",
    "fromAmino": [Function],
    "toAmino": [Function],
  },
  "/osmosis.lockup.MsgLockTokens": Object {
    "aminoType": "osmosis/lockup/lock-tokens",
    "fromAmino": [Function],
    "toAmino": [Function],
  },
  "/osmosis.superfluid.MsgLockAndSuperfluidDelegate": Object {
    "aminoType": "osmosis/lock-and-superfluid-delegate",
    "fromAmino": [Function],
    "toAmino": [Function],
  },
  "/osmosis.superfluid.MsgSuperfluidDelegate": Object {
    "aminoType": "osmosis/superfluid-delegate",
    "fromAmino": [Function],
    "toAmino": [Function],
  },
  "/osmosis.superfluid.MsgSuperfluidUnbondLock": Object {
    "aminoType": "osmosis/superfluid-unbond-lock",
    "fromAmino": [Function],
    "toAmino": [Function],
  },
  "/osmosis.superfluid.MsgSuperfluidUndelegate": Object {
    "aminoType": "osmosis/superfluid-undelegate",
    "fromAmino": [Function],
    "toAmino": [Function],
  },
  "/osmosis.superfluid.MsgUnPoolWhitelistedPool": Object {
    "aminoType": "osmosis/unpool-whitelisted-pool",
    "fromAmino": [Function],
    "toAmino": [Function],
  },
}
`;

exports[`lockTokens.fromAmino 1`] = `
Object {
  "coins": Array [
    Object {
      "amount": "1000",
      "denom": "mydenom",
    },
  ],
  "duration": Object {
    "nanos": 0,
    "seconds": "100",
  },
  "owner": "osmo1addresshere",
}
`;

exports[`lockTokens.toAmino 1`] = `
Object {
  "coins": Array [
    Object {
      "amount": "1000",
      "denom": "mydenom",
    },
  ],
  "duration": Object {
    "nanos": 0,
    "seconds": "100",
  },
  "owner": "osmo1addresshere",
}
`;
